From f8e559fbab6353d5b129e8127e2e38e7385c8e4b Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Tue, 11 Sep 2018 17:00:14 +0300
Subject: [PATCH 298/706] enetc: Fall back to PHY-less operation if of phy node
 missing

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit ac156ae455abb999d0ac5b43bf0f285fc68c4897)
(cherry picked from commit 51a51806765d00f282da9de1521658a8937fef93)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index 1d5c2d8620d0..d011e2b94a30 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -805,7 +805,7 @@ static int enetc_pf_probe(struct pci_dev *pdev,
 
 	err = enetc_of_get_phy(priv);
 	if (err)
-		goto err_of_get_phy;
+		dev_warn(&pdev->dev, "Fallback to PHY-less operation\n");
 
 	netif_carrier_off(ndev);
 
@@ -814,8 +814,6 @@ static int enetc_pf_probe(struct pci_dev *pdev,
 
 	return 0;
 
-err_of_get_phy:
-	enetc_free_irqs(priv);
 err_setup_irq:
 	unregister_netdev(ndev);
 err_reg_netdev:
@@ -846,9 +844,10 @@ static void enetc_pf_remove(struct pci_dev *pdev)
 		   enetc_drv_name, enetc_drv_ver);
 	unregister_netdev(si->ndev);
 
-	if (of_phy_is_fixed_link(priv->dev->of_node))
-		of_phy_deregister_fixed_link(priv->dev->of_node);
-	of_node_put(priv->phy_node);
+	if (pdev->dev.of_node && of_phy_is_fixed_link(pdev->dev.of_node))
+		of_phy_deregister_fixed_link(pdev->dev.of_node);
+	if (priv->phy_node)
+		of_node_put(priv->phy_node);
 
 	enetc_free_irqs(priv);
 	enetc_free_msix(priv);
-- 
2.17.1

